tests/Findologic/Request/Parser/NavigationCategoryParserTest.php

```diff
@@ -0,0 +1,129 @@
+    }
+
+    public function testFindsCategoryFromPage(): void
+    {
+        $expectedCategory = $this->getCategory();
+        $this->genericPageLoader = $this->getMockBuilder(GenericPageLoader::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $navigationTreeMock = $this->getMockBuilder(Tree::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $navigationTreeMock->expects($this->any())->method('getActive')->willReturn($expectedCategory);
+
+        /** @var HeaderPagelet|MockObject $headerPageletMock */
+        $headerPageletMock = $this->getMockBuilder(HeaderPagelet::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $headerPageletMock->expects($this->any())->method('getNavigation')->willReturn($navigationTreeMock);
+
+        $page = new Page();
+        $page->setHeader($headerPageletMock);
+
+        $this->genericPageLoader->expects($this->once())->method('load')->willReturn($page);
+
+        $request = new Request();
+        $salesChannelContext = $this->buildSalesChannelContext();
+        $category = $this->getDefaultNavigationCategoryParser()->parse($request, $salesChannelContext);
+
+        $this->assertEquals($expectedCategory, $category);
+    }
+
+    public function testSalesChannelContextIsNotOverriddenByInternalPageRequest(): void
+    {
+        $expectedCategory = $this->getCategory();
+        $this->genericPageLoader = $this->getMockBuilder(GenericPageLoader::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $navigationTreeMock = $this->getMockBuilder(Tree::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $navigationTreeMock->expects($this->any())->method('getActive')->willReturn($expectedCategory);
+
+        /** @var HeaderPagelet|MockObject $headerPageletMock */
+        $headerPageletMock = $this->getMockBuilder(HeaderPagelet::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $headerPageletMock->expects($this->any())->method('getNavigation')->willReturn($navigationTreeMock);
+
+        $page = new Page();
+        $page->setHeader($headerPageletMock);
+
+        $request = new Request();
+        $salesChannelContext = $this->buildSalesChannelContext();
+
+        $this->genericPageLoader->expects($this->once())->method('load')
+            ->with($request, $salesChannelContext)
+            ->willReturnCallback(function ($request, $actualSalesChannelContext) use ($page, $salesChannelContext) {
+                // Ensure that used instance is not the same. This prevents Shopware from modifying it.
+                $this->assertNotSame($salesChannelContext, $actualSalesChannelContext);
+
+                return $page;
+            });
+
+        $this->getDefaultNavigationCategoryParser()->parse($request, $salesChannelContext);
+    }
+
+    private function getCategory(): CategoryEntity
```

Comment:
Set `$this->defaultCategory` in setup?

Reply:
I chose against it, since I do not want to create entities in the database by default.